src/controller/c/camera.c

```diff
@@ -707,8 +707,7 @@ const unsigned char *wb_camera_get_image(WbDeviceTag tag) {
     return ac->image->data;
 
   robot_mutex_lock_step();
-  bool success = image_request(ac->image, __FUNCTION__);
-  if (!ac->image->data || !success) {
+  if (!ac->image->data) {
```

Comment:
Tried here 691dc1b (Lidar and range finders)

The implementation of rotational lidars is convoluted, and I cannot make them work. It crashes and it doesn't give a correct result. It is probably due to the lifecycle of the device that I don't fully understand.

The `image_request` is used by the image segmentation, so I should check with @stefaniapedrazzi why is it used there.

Let me know if I should invest more time.

Reply:
I believe we should apply the same mechanism to the segmented image as well.
As for the Lidar, I understand the rotational aspect of Lidar may induce some extra care, but in principle, it could work the same way as normal cameras when it comes to sending the result to the controller.
Also, the range-finder should be pretty similar to the camera.
I believe it is important to invest more time on this to be sure we are doing the best performance wise.
A very useful test would be to verify that we don't write twice the camera image in the shared memory segment (`copyImageToSharedMemory`) during the same step.